spectrocloud/resource_cluster_tke.go

```diff
@@ -0,0 +1,764 @@
+package spectrocloud
+
+import (
+	"context"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/spectrocloud/gomi/pkg/ptr"
+	"github.com/spectrocloud/hapi/models"
+	"github.com/spectrocloud/terraform-provider-spectrocloud/pkg/client"
+)
+
+func resourceClusterTke() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceClusterTkeCreate,
+		ReadContext:   resourceClusterTkeRead,
+		UpdateContext: resourceClusterTkeUpdate,
+		DeleteContext: resourceClusterDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		SchemaVersion: 2,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"tags": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Set:      schema.HashString,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"cluster_profile_id": {
```

Comment:
@nikchern, let's remove `cluster_profile_id` from edge, libvirt, vsphere_edge too as it is deprecated
Line 46-50